card/lib/card/env.rb

```diff
@@ -5,40 +5,40 @@ module Env
     class << self
       def reset args={}
         @@env = { :main_name => nil }
-        
+
         if c = args[:controller]
           self[:controller] = c
           self[:session]    = c.request.session
-          self[:params]     = c.request.params
+          self[:params]     = c.params
```

Comment:
I don't know what the difference is between  `request.params` and `params` but changes to `request.params` don't affect `params` anymore. In card_controller we use `params` so we have to use here `params` too.